AppCenter/AppCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -264,100 +275,172 @@ + (void)enableAutoVacuumInOpenedDatabase:(void *)db {
-
-    // Loop on rows.
-    while (sqlite3_step(statement) == SQLITE_ROW) {
-      NSMutableArray *entry = [NSMutableArray new];
-
-      // Loop on columns.
-      for (int i = 0; i < sqlite3_column_count(statement); i++) {
-        id value = nil;
-
-        /*
-         * Convert values.
-         * TODO: Add here any other type it needs.
-         */
-        switch (sqlite3_column_type(statement, i)) {
-        case SQLITE_INTEGER:
-          value = @(sqlite3_column_int(statement, i));
-          break;
-        case SQLITE_TEXT:
-          value = [NSString stringWithUTF8String:(const char *)sqlite3_column_text(statement, i)];
-          break;
-        default:
-          value = [NSNull null];
-          break;
-        }
-        [entry addObject:value];
-      }
-      if (entry.count > 0) {
-        [entries addObject:entry];
-      }
-    }
-    sqlite3_finalize(statement);
-  } else {
-    MSLogError([MSAppCenter logTag], @"Query \"%@\" failed with error: %d - %@", query, prepareResult,
-               [NSString stringWithUTF8String:sqlite3_errmsg(db)]);
+  int queryResult = [MSDBStorage executeQuery:query
+                             inOpenedDatabase:db
+                                   withValues:values
+                                   usingBlock:^(void *statement) {
+                                     int stepResult;
+
+                                     // Loop on rows.
+                                     while ((stepResult = sqlite3_step(statement)) == SQLITE_ROW) {
+                                       NSMutableArray *entry = [NSMutableArray new];
+
+                                       // Loop on columns.
+                                       for (int i = 0; i < sqlite3_column_count(statement); i++) {
+                                         NSObject *value = [MSDBStorage columnValueFromStatement:statement withIndex:i];
+                                         [entry addObject:value];
+                                       }
+                                       if (entry.count > 0) {
+                                         [entries addObject:entry];
+                                       }
+                                     }
+                                     if (stepResult != SQLITE_DONE) {
+                                       NSString *errorMessage = [NSString stringWithUTF8String:sqlite3_errmsg(db)];
+                                       MSLogError([MSAppCenter logTag], @"Query failed with error: %d\n\t%@", stepResult, errorMessage);
+                                       return stepResult;
+                                     }
+                                     return SQLITE_OK;
+                                   }];
+  if (result) {
+    *result = queryResult;
   }
   return entries;
 }
 
++ (NSObject *)columnValueFromStatement:(sqlite3_stmt *)statement withIndex:(int)index {
+
+  /*
+   * Convert values.
+   * TODO: Add here any other type it needs.
```

Comment:
Is this a real TODO? Like we know there is more that needs to be here at this moment? Or is it just the place to add types in the future?

Reply:
It was just moved from another place, not a new thing. My vote to remove this TODO